monai/handlers/mlflow_handler.py

```diff
@@ -115,14 +183,61 @@ def attach(self, engine: Engine) -> None:
             engine.add_event_handler(Events.ITERATION_COMPLETED, self.iteration_completed)
         if self.epoch_log and not engine.has_event_handler(self.epoch_completed, Events.EPOCH_COMPLETED):
             engine.add_event_handler(Events.EPOCH_COMPLETED, self.epoch_completed)
+        if not engine.has_event_handler(self.complete, Events.COMPLETED):
+            engine.add_event_handler(Events.COMPLETED, self.complete)
 
-    def start(self) -> None:
+    def start(self, engine: Engine) -> None:
         """
         Check MLFlow status and start if not active.
 
         """
+        mlflow.set_experiment(self.experiment_name)
         if mlflow.active_run() is None:
-            mlflow.start_run()
+            run_name = f"run_{time.strftime('%Y%m%d_%H%M%S')}" if self.run_name is None else self.run_name
+            mlflow.start_run(run_name=run_name)
+
+        if self.experiment_param:
+            # avoid to recording same parameters in the same run
+            self._delete_exist_param_in_dict(self.experiment_param)
```

Comment:
I don't quite understand why we need to delete the existing values?
I think the values in `experiment_param` are const, right?
And the same engine object may `start -> end` several times during training.

Thanks.

Reply:
I think we don't suppose to attach the same handler object to more than 1 engine, do we still need to worry about this problem?

Reply:
If it can be ensured that the `experiment_param` in trainer and validator won't have any same key, then we can remove this part.
Thanks,
Bin